src/test/java/net/openhft/chronicle/wire/MethodWriterVagueTypesTest.java

```diff
@@ -67,65 +69,90 @@ public int hashCode() {
+                    singleQ.add(message);
+                });
         // test with a series of objects
         testSingle(printer, reader, new MarshallableTestContainer());
         testSingle(printer, reader, "hello");
         testSingle(printer, reader, new NonMarshallableTestContainer());
         testSingle(printer, reader, new MarshallableTestContainer());
         testSingle(printer, reader, new NonMarshallableTestContainer());
+    }
+
+    @Test
+    public void testDouble() throws Exception {
+        // Initialization of the wire
+        Wire w = new TextWire(Bytes.allocateElasticOnHeap());
+        PrintObjectDouble printer = w.methodWriter(PrintObjectDouble.class);
 
+        // Set up a MethodReader to read the String message and process it using the println method
+        MethodReader reader = w.methodReaderBuilder()
+                .multipleNonMarshallableParamTypes(multipleNonMarshallableParamTypes)
+                .build((PrintObjectDouble) (key, message) -> {
+                    doubleQ.add(key);
+                    doubleQ.add(message);
+                });
+        // test with a series of objects
         testDouble(printer, reader, "key1", new MarshallableTestContainer());
         testDouble(printer, reader, "key2", "value2");
         testDouble(printer, reader, "key3", new NonMarshallableTestContainer());
         testDouble(printer, reader, new MarshallableTestContainer(), new MarshallableTestContainer());
         testDouble(printer, reader, new NonMarshallableTestContainer(), new NonMarshallableTestContainer());
     }
 
-    private void testSingle(PrintObject printer, MethodReader reader, Object obj) throws Exception {
-        test(() -> printer.singleMsg(obj), reader, singleQ, obj);
+    private void testSingle(PrintObjectSingle printer, MethodReader reader, Object obj) throws Exception {
+        test(() -> printer.msg(obj), reader, singleQ, obj);
     }
 
-    private void testDouble(PrintObject printer, MethodReader reader, Object key, Object obj) throws Exception {
-        test(() -> printer.doubleMsg(key, obj), reader, doubleQ, key, obj);
+    private void testDouble(PrintObjectDouble printer, MethodReader reader, Object key, Object obj) throws Exception {
+        test(() -> printer.msg(key, obj), reader, doubleQ, key, obj);
     }
 
     private void test(Runnable methodCall, MethodReader reader, ArrayBlockingQueue<Object> queue, Object ... objs) throws Exception {
         methodCall.run();
+        boolean marshallableToNonMarshallable = false;
+        if (prevObjClasses[0] != null) {
+            for (int i=0; i < objs.length; i++) {
+                marshallableToNonMarshallable |= Marshallable.class.isAssignableFrom(prevObjClasses[i]) && !(objs[i] instanceof Marshallable);
+            }
+        }
+        if (Boolean.FALSE.equals(multipleNonMarshallableParamTypes) && marshallableToNonMarshallable) {
+            Assert.assertThrows(Exception.class, () -> assertWrite(reader, queue, objs));
```

Comment:
cool

Reply:
what exception does it throw?

Reply:
In this scenario, it either throws a RuntimeException from `assertWrite()` or an `IORuntimeException` is thrown by TextWire while marshalling - I can change it to a RuntimeException